protect data from unauthorized users #169419213

Muhire-Josue opened the pull request:
#### What does this PR do?

- It blocks unauthorized users to view an entry

#### Description of Task to be completed?

- Users should not view other user's entries, there should be restricted access.

#### How should this be manually tested?

- Use this link in postman: https://mydiary-api-host.herokuapp.com/api/v1/entries/:userId

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [169419213](https://www.pivotaltracker.com/story/show/169419213)

#### Screenshots (if appropriate)
![forbiden](https://user-images.githubusercontent.com/37122177/67759837-dcc50700-fa48-11e9-8771-a82f2ef08b0c.PNG)
![view-diary](https://user-images.githubusercontent.com/37122177/67759838-dcc50700-fa48-11e9-9b89-9a5d5718b2fa.PNG)
![permission](https://user-images.githubusercontent.com/37122177/67765057-e99b2800-fa53-11e9-81b0-bec2c4b8900f.PNG)

Muhire-Josue committed:
chore(protect data from unauthorized users)

	- add authenticate users who view an entry

[Finishes #169419213]

TheoOkafor commented on server/controllers/entryController.js:
```diff
@@ -46,7 +46,10 @@ class entryController {
       return res.status(404).json({ status: 404, error: 'Entry not found' });
     }
-    return res.status(200).json({ status: 200, data: entry });
+    if (entry.userId === parseInt(req.user.id)) {
```
Since you are going to use this permission check for PATCH, DELETE, GET, make this a middleware. You can call the middleware `permissions` or any name of your choice.

TheoOkafor replied:
implement the middleware in the endpoints were it's needed. That way you won't have repeated codes in your codebase

Muhire-Josue replied:
Done

Muhire-Josue committed:
chore(permission middleware): add permission middleware

	- add middleware to test the id provided by the user
	- add also test if user should have access to data
